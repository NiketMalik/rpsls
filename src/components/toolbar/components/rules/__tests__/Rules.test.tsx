import React from "react";
import {
  fireEvent,
  render,
  screen,
  waitFor,
  waitForElementToBeRemoved,
} from "test-utils";

import { Rules } from "../";

describe("Rules", () => {
  it("renders without error", () => {
    const { container } = render(<Rules />);
    const modalTriggerEle = screen.getByTestId("rules-modal-trigger");
    expect(modalTriggerEle).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  describe("tooltip actions", () => {
    it("should show tooltip", async () => {
      render(<Rules />);

      const modalTriggerEle = screen.getByTestId("rules-modal-trigger", {});
      expect(modalTriggerEle).toBeInTheDocument();

      fireEvent.mouseOver(modalTriggerEle);

      await waitFor(() => {
        const triggerTooltipEle = screen.getByTestId(
          "rules-modal-trigger-tooltip",
        );
        expect(triggerTooltipEle).toBeInTheDocument();
        expect(triggerTooltipEle).toMatchSnapshot();
      });
    });
  });

  describe("modal actions", () => {
    it("should open modal on trigger open", async () => {
      render(<Rules />);

      const modalTriggerEle = screen.getByTestId("rules-modal-trigger", {});
      expect(modalTriggerEle).toBeInTheDocument();

      fireEvent.click(modalTriggerEle);

      await waitFor(() => {
        const modalEle = screen.getByTestId("rules-modal");
        expect(modalEle).toBeInTheDocument();
        expect(modalEle).toMatchSnapshot();

        const modalContentEle = screen.getByTestId("rules-modal-content");
        expect(modalContentEle).toBeInTheDocument();
        expect(modalContentEle).toMatchSnapshot();
      });
    });

    it("should close modal on click outside", async () => {
      const { container } = render(<Rules />);

      const modalTriggerEle = screen.getByTestId("rules-modal-trigger", {});
      expect(modalTriggerEle).toBeInTheDocument();

      fireEvent.click(modalTriggerEle);

      await waitFor(() => {
        const modalEle = screen.getByTestId("rules-modal");
        expect(modalEle).toBeInTheDocument();
      });

      fireEvent.keyDown(container, {
        key: "Escape",
        code: "Escape",
        keyCode: 27,
        charCode: 27,
      });

      await waitForElementToBeRemoved(() => screen.getByTestId("rules-modal"));
      expect(container).toMatchSnapshot();
    });
  });
});
