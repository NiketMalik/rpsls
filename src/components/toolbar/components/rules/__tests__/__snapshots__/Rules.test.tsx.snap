// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rules modal actions should close modal on click outside 1`] = `
<div>
  <button
    aria-label="rules"
    class="btn btn-trasnparent"
    data-testid="rules-modal-trigger"
    type="button"
  >
    <img
      alt="rules"
      class=""
      height="32"
      src="icon-instructions.svg"
      width="32"
    />
  </button>
</div>
`;

exports[`Rules modal actions should open modal on trigger open 1`] = `
<div
  class="modal-dialog"
  data-testid="rules-modal"
>
  <div
    class="modal-content"
  />
</div>
`;

exports[`Rules modal actions should open modal on trigger open 2`] = `
<div
  class="modal-dialog"
  data-testid="rules-modal"
>
  <div
    class="modal-content"
  >
    <div
      class="modal-body"
      data-testid="rules-modal-content"
    >
      <div
        class="d-flex flex-column align-items-center"
      >
        <div
          class="display-6"
        >
          Rules
        </div>
        <img
          alt="rules"
          class="p-4 img-fluid"
          src="rules.svg"
          width="100%"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Rules modal actions should open modal on trigger open 3`] = `
<div
  class="modal-body"
  data-testid="rules-modal-content"
>
  <div
    class="d-flex flex-column align-items-center"
  >
    <div
      class="display-6"
    >
      Rules
    </div>
    <img
      alt="rules"
      class="p-4 img-fluid"
      src="rules.svg"
      width="100%"
    />
  </div>
</div>
`;

exports[`Rules renders without error 1`] = `
<div>
  <button
    aria-label="rules"
    class="btn btn-trasnparent"
    data-testid="rules-modal-trigger"
    type="button"
  >
    <img
      alt="rules"
      class=""
      height="32"
      src="icon-instructions.svg"
      width="32"
    />
  </button>
  <div />
</div>
`;

exports[`Rules tooltip actions should show tooltip 1`] = `
<div
  class="fade show tooltip bs-tooltip-top"
  data-testid="rules-modal-trigger-tooltip"
  id="toolbar-rules-tooltip"
  role="tooltip"
  x-placement="top"
>
  <div
    class="tooltip-arrow"
  />
  <div
    class="tooltip-inner"
  >
    Game rules
  </div>
</div>
`;
